### PR TITLE
Miner fixes

### PR DESCRIPTION
**Pull request by turuslan**
- ask api
- fix double padded/unpadded conversion
- fix buffer overflow
- thread for retrieval unseal
- retrieval persistent config
- precommit policy params order
- add market providers to miner main
- fix win count poly coefficients
- fix window post uninitialized proof type
- sealing
    - start packing on full sector
    - fix last piece didn't fit sector
    - fix min/max in expiration
- update proof params url
- fix proofs not scheduling for single worker
- car reader detect zero padded file

### CODE REVIEW

**Review by Alexey-N-Chernyshov — APPROVED** [review]

**Review by Alexey-N-Chernyshov — APPROVED** [review]

File: core/miner/main/main.cpp [hunk 1]
```diff
@@ -328,14 +340,94 @@ namespace fc {
...
+            napi,
+            piece_storage,
+            markets_ipld,
+            one_key("retrieval_provider_ask", leveldb),
+            manager,
+            miner)};
+    retrieval_provider->start();
+
     auto mapi{std::make_shared<api::Api>()};
+    mapi->DealsImportData = [&](auto &proposal, auto &path) {
```
Comment (ortyomka) [inline core/miner/main/main.cpp:394]: Maybe we should move api initiation in another file, as it is with node api

File: core/miner/mining.cpp [hunk 2]
```diff
@@ -211,7 +211,11 @@ namespace fc::mining {
                           const BigInt &total_power) {
     constexpr auto P{256};
     static auto poly{[P](std::initializer_list<std::string> s) {
-      return [P, p{std::vector<BigInt>{s.begin(), s.end()}}](const BigInt &x) {
+      std::vector<BigInt> p{s.begin(), s.end()};
```
Comment (ortyomka) [inline core/miner/mining.cpp:214]: Maybe choose more readable names

File: core/proofs/impl/proofs.cpp [hunk 3]
```diff
@@ -1040,10 +1040,10 @@ namespace fc::proofs {
 
     uint64_t left = piece_size;
     constexpr auto kDefaultBufferSize = uint64_t(32 * 1024);
-    std::vector<uint8_t> buffer(kDefaultBufferSize);
     auto chunks = kDefaultBufferSize / 127;
     PaddedPieceSize outTwoPow =
         primitives::piece::paddedSize(chunks * 128).padded();
+    std::vector<uint8_t> buffer(outTwoPow.unpadded());
```
Comment (ortyomka) [inline core/proofs/impl/proofs.cpp:1046]: Already fixed

File: core/proofs/proofs.hpp [hunk 4]
```diff
@@ -280,6 +280,8 @@ namespace fc::proofs {
    private:
     static fc::common::Logger logger_;
   };
+
+  UnpaddedPieceSize padPiece(const std::string &path);
```
Comment (ortyomka) [inline core/proofs/proofs.hpp:284]: Maybe move it to up, because can be lost under the huge class

Comment (turuslan) [reply core/proofs/proofs.hpp:284]: We already have `proofs` namespace, `Proofs` static methods are like namespace

File: core/proofs/impl/proofs.cpp (hunk 3 above)
Comment (turuslan) [reply core/proofs/impl/proofs.cpp:1046]: No

File: core/miner/main/main.cpp (hunk 1 above)
Comment (turuslan) [reply core/miner/main/main.cpp:394]: It's miner api

File: core/proofs/proofs.hpp (hunk 4 above)
Comment (ortyomka) [reply core/proofs/proofs.hpp:283]: Just for readability

**Review by ortyomka — APPROVED** [review]

**Pull request merged**